### ISSUE TITLE
Incomplete unnecessary decomposition of U+0268 LATIN SMALL LETTER I WITH STROKE

### ISSUE THREAD

**Issue by dscorbett** [googlefonts/noto-fonts#1810]
### Font
NotoMono-Regular.otf

### Where the font came from, and when
Site: https://github.com/googlefonts/noto-fonts/blob/bf20559450ec75aec7a646b208343540a4496262/phaseIII_only/unhinted/otf/NotoMono/NotoMono-Regular.otf
Date: 2020-08-01

### Font version
Version 2.004

### Issue
The glyph for U+0268 LATIN SMALL LETTER I WITH STROKE is substituted with the glyphs `i` and `uni0335` (except when the language system tag is APPH or IPPH). This makes the stroke appear longer than it was designed to be.

### Character data
ɨ
U+0268 LATIN SMALL LETTER I WITH STROKE

### Screenshot
Default:
![ɨ](https://user-images.githubusercontent.com/1124347/89733587-5f411500-da24-11ea-9287-35aef3f9b513.png)
Language system tag APPH or IPPH:
![ɨ](https://user-images.githubusercontent.com/1124347/89733588-5f411500-da24-11ea-9c05-50a5ab3bfadf.png)

**Comment by nizarsq** [comment]
<img width="753" alt="Screen Shot 2020-08-20 at 2 14 39 PM" src="https://user-images.githubusercontent.com/22158882/90826548-7d91f500-e2ef-11ea-9d45-4733f0efcf92.png">
<img width="501" alt="Screen Shot 2020-08-20 at 2 14 58 PM" src="https://user-images.githubusercontent.com/22158882/90826573-88e52080-e2ef-11ea-980a-41ff080ecb86.png">
<img width="879" alt="Screen Shot 2020-08-20 at 2 15 10 PM" src="https://user-images.githubusercontent.com/22158882/90826588-900c2e80-e2ef-11ea-876c-e80c315550b2.png">

**Comment by verdy-p** [comment]
Why do we even need such variants ? IMHO the long-stroke variant is unnecessary and ugly (I barely see any reason for creating a confusion opportunity when such long strong would join with the stroke of another letter.
The short-stroke variant (undecomped glyph) should just be the default (not even needing the AAPH or IPPH language feature tag) as it is also visually better and enforces the "single letter" semantic (even if it is not used for phonetic notations).
This is just like if we needed to compose a latin capital "F" with a gamma and a stroke diacritic.

And the decomposed glyphs with a long-stroke diacritic, if it is used, seems to be incorrectly positioned (not correctly centered, too much to the right, so it would not even join horizontally with a stroke on another letter: just try composing two U+0268 side by side, the second one does not join with the first one: this should still be unjoined and using short strokes (shorter than the top and bottom serifs, or shorter than normal advance gaps if there are not serifs) centered almost equally around the vertical bar, except possibly for italic/oblique or cursive styles where there are visual artefacts that may be needed for correct blackness at very small font sizes (where serifs could be significantly modified/reduced/removed, and some adaptations to avoid creating new confusables, e.g. with a lower case "r", "t" or "f" especially with italic???).

**Comment by moyogo** [comment]
I don’t think this is intentional, the substitution or what language-systems it ends up in.

There is a uni0268dotless which should have been used in a contextual lookup whenever uni0268 is followed by a combining mark above.

Glyphs.app’s automatic ccmp does something similar to what the issue is, if there’s uni0268, i and strokeshortcomb it will write a lookup that decomposes uni0268 as i strokeshortcomb. I think it does the same with the Cyrillic istroke. This wrong in most cases.
(/cc @schriftgestalt)

**Comment by schriftgestalt** [comment]
You are right. I remove it.

**Comment by verdy-p** [comment]
Good to know. At least the cause is known and will help avoid similar errors in the future. This behavior can only be useful as a "last chance" fallback to help get better coverage and usability of fonts (when there's no specific mapping already and no reason to focus only specific "languages/notations" and no known indication that this is only specific to these and not applicable at all in the general case).

**Comment by moyogo** [comment]
> You are right. I remove it (the one for uni0268, I don’t think the cyrillic was touched).

@schriftgestalt You are right, I got confused with some other issue with Noto and i-cy. There is no istroke-cy.

**Comment by verdy-p** [comment]
anyway this bug is not very critical and simple to fix on this specific fix (just a bit harder to avoid in its generic case for the tools that caused it).
Thge serious issues are about really incorrect and confusing glyphs that exist for some scripts even in their base alphabet, or very incorrect placement of diacritics which are unreadable, forcing the reader to also guess with his knowledge, possibly wrong, and causing accessibility problems or to enter non-stadnard tweaked texts

**Comment by schriftgestalt** [comment]
> There is no istroke-cy.

There is a gestroke-cy. But that is not touched.
